datafusion/sql/src/statement.rs

```diff
@@ -127,42 +129,66 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 && table_properties.is_empty()
                 && with_options.is_empty() =>
             {
-                let plan = self.query_to_plan(*query, planner_context)?;
-                let input_schema = plan.schema();
-
-                let plan = if !columns.is_empty() {
-                    let schema = self.build_schema(columns)?.to_dfschema_ref()?;
-                    if schema.fields().len() != input_schema.fields().len() {
-                        return Err(DataFusionError::Plan(format!(
+                match query {
+                    Some(query) => {
```

Comment:
looks like this code will handle traditoinal create table no just CTAS, do you think we can remove ```
            Statement::CreateTable { .. } => Err(DataFusionError::NotImplemented(
                "Only `CREATE TABLE table_name AS SELECT ...` statement is supported"
                    .to_string(),
            )),
            ```
            condition?